cumulus/pallets/xcmp-queue/src/lib.rs

```diff
@@ -491,7 +491,7 @@ impl<T: Config> Pallet<T> {
 		let channel_info =
 			T::ChannelInfo::get_channel_info(recipient).ok_or(MessageSendError::NoChannel)?;
 		// Max message size refers to aggregates, or pages. Not to individual fragments.
-		let max_message_size = channel_info.max_message_size as usize;
+		let max_message_size = channel_info.max_message_size.min(T::MaxPageSize::get()) as usize;
```

Comment:
```suggestion
		let max_message_size = channel_info.max_message_size.min.saturating_add(XcmpMessageFormat::max_encoded_len())(T::MaxPageSize::get()) as usize;
```

The `XcmpMessageFormat` is part  of the `page`. So, if we don't do this, we will never be able to storage a message of `max_message_size` (when this is smaller than `T::MaxPageSize`).

Reply:
Yes, but i think this is a hard limit from the relay chain that we cannot go over. So the "max message size" is for a HRMP message - not for the XCM inside of it.